// react-native-paper-dates/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Calendar: true,
  DatePickerModal: true,
  DatePickerModalContent: true,
  TimePickerModal: true,
  TimePicker: true,
  DatePickerInput: true,
  registerTranslation: true,
  getTranslation: true,
  ar: true,
  de: true,
  en: true,
  enGB: true,
  es: true,
  fr: true,
  he: true,
  hi: true,
  it: true,
  ko: true,
  nl: true,
  pl: true,
  pt: true,
  tr: true
};
Object.defineProperty(exports, "Calendar", {
  enumerable: true,
  get: function () {
    return _Calendar.default;
  }
});
Object.defineProperty(exports, "DatePickerInput", {
  enumerable: true,
  get: function () {
    return _DatePickerInput.default;
  }
});
Object.defineProperty(exports, "DatePickerModal", {
  enumerable: true,
  get: function () {
    return _DatePickerModal.default;
  }
});
Object.defineProperty(exports, "DatePickerModalContent", {
  enumerable: true,
  get: function () {
    return _DatePickerModalContent.default;
  }
});
Object.defineProperty(exports, "TimePicker", {
  enumerable: true,
  get: function () {
    return _TimePicker.default;
  }
});
Object.defineProperty(exports, "TimePickerModal", {
  enumerable: true,
  get: function () {
    return _TimePickerModal.default;
  }
});
Object.defineProperty(exports, "ar", {
  enumerable: true,
  get: function () {
    return _ar.default;
  }
});
Object.defineProperty(exports, "de", {
  enumerable: true,
  get: function () {
    return _de.default;
  }
});
Object.defineProperty(exports, "en", {
  enumerable: true,
  get: function () {
    return _en.default;
  }
});
Object.defineProperty(exports, "enGB", {
  enumerable: true,
  get: function () {
    return _enGB.default;
  }
});
Object.defineProperty(exports, "es", {
  enumerable: true,
  get: function () {
    return _es.default;
  }
});
Object.defineProperty(exports, "fr", {
  enumerable: true,
  get: function () {
    return _fr.default;
  }
});
Object.defineProperty(exports, "getTranslation", {
  enumerable: true,
  get: function () {
    return _utils.getTranslation;
  }
});
Object.defineProperty(exports, "he", {
  enumerable: true,
  get: function () {
    return _he.default;
  }
});
Object.defineProperty(exports, "hi", {
  enumerable: true,
  get: function () {
    return _hi.default;
  }
});
Object.defineProperty(exports, "it", {
  enumerable: true,
  get: function () {
    return _it.default;
  }
});
Object.defineProperty(exports, "ko", {
  enumerable: true,
  get: function () {
    return _ko.default;
  }
});
Object.defineProperty(exports, "nl", {
  enumerable: true,
  get: function () {
    return _nl.default;
  }
});
Object.defineProperty(exports, "pl", {
  enumerable: true,
  get: function () {
    return _pl.default;
  }
});
Object.defineProperty(exports, "pt", {
  enumerable: true,
  get: function () {
    return _pt.default;
  }
});
Object.defineProperty(exports, "registerTranslation", {
  enumerable: true,
  get: function () {
    return _utils.registerTranslation;
  }
});
Object.defineProperty(exports, "tr", {
  enumerable: true,
  get: function () {
    return _tr.default;
  }
});

var _Calendar = _interopRequireDefault(require("./Date/Calendar"));

var _DatePickerModal = _interopRequireWildcard(require("./Date/DatePickerModal"));

Object.keys(_DatePickerModal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _DatePickerModal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _DatePickerModal[key];
    }
  });
});

var _DatePickerModalContent = _interopRequireDefault(require("./Date/DatePickerModalContent"));

var _TimePickerModal = _interopRequireDefault(require("./Time/TimePickerModal"));

var _TimePicker = _interopRequireDefault(require("./Time/TimePicker"));

var _DatePickerInput = _interopRequireDefault(require("./Date/DatePickerInput"));

var _utils = require("./translations/utils");

var _ar = _interopRequireDefault(require("./translations/ar"));

var _de = _interopRequireDefault(require("./translations/de"));

var _en = _interopRequireDefault(require("./translations/en"));

var _enGB = _interopRequireDefault(require("./translations/enGB"));

var _es = _interopRequireDefault(require("./translations/es"));

var _fr = _interopRequireDefault(require("./translations/fr"));

var _he = _interopRequireDefault(require("./translations/he"));

var _hi = _interopRequireDefault(require("./translations/hi"));

var _it = _interopRequireDefault(require("./translations/it"));

var _ko = _interopRequireDefault(require("./translations/ko"));

var _nl = _interopRequireDefault(require("./translations/nl"));

var _pl = _interopRequireDefault(require("./translations/pl"));

var _pt = _interopRequireDefault(require("./translations/pt"));

var _tr = _interopRequireDefault(require("./translations/tr"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
